test_log/src/main/java/org/corfudb/test_log/TestLog.java

```diff
@@ -0,0 +1,289 @@
+                }
+                f_strm_i.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private static void processLogFile(FileChannel fc_in,
+                                       FileChannel fc_out,
+                                       Operation op) throws IOException {
+        boolean display = op.getOpType() == Operation.OperationType.DISPLAY ||
+                    op.getOpType() == Operation.OperationType.DISPLAY_ALL;
+        boolean display_data = op.getOpType() == Operation.OperationType.DISPLAY_ALL;
+
+        System.out.format("processLogFile display %d output %d\n", display ? 1 : 0,
+                fc_out != null ? 1 : 0);
+
+        // Metadata
+        //   LogHeader
+        // Metadata
+        //   LogEntry
+        // ...
+        fc_in.position(0);
+        long file_size = fc_in.size();
+        ByteBuffer md_buffer = ByteBuffer.allocate(MetadataSize);
+        int r = fc_in.read(md_buffer);
+        md_buffer.flip();
+        if (fc_out != null) {
+            fc_out.write(md_buffer);
+        }
+        if (r > 0) {
+            Metadata md = Metadata.parseFrom(md_buffer.array());
+            if (display) {
+                System.out.println("checksum " + String.format("%08x", md.getChecksum()));
+                System.out.println("length " + Integer.toString(md.getLength()));
+            }
+            ByteBuffer lh_buffer = ByteBuffer.allocate(LogHeaderSize);
+            r = fc_in.read(lh_buffer);
+            lh_buffer.flip();
+            if (fc_out != null) {
+                fc_out.write(lh_buffer);
+            }
+            if (r > 0) {
+                LogHeader lh = LogHeader.parseFrom(lh_buffer.array());
+                if (display) {
+                    System.out.println("version " + Integer.toString(lh.getVersion()));
+                    System.out.println("verify " + Boolean.toString(lh.getVerifyChecksum()));
+                }
+            }
+        }
+        HashFunction hf = Hashing.crc32c();
+        long rem_size = fc_in.size() - fc_in.position();  // if size == position then file pointer is off the end
+        int record_cnt = 0;
+        boolean replace_entry = false;
+        boolean enable_write = false;
+        switch (op.getOpType()) {
+            case ERASE_ONE:
+            case ERASE_REMAINING:
+                enable_write = true;
+                break;
+            default:
+                enable_write = false;
+        }
+        while (rem_size > 0) {
+            boolean trigger = false;
+            ByteBuffer comma_buffer = ByteBuffer.allocate(2);
+            int bytes_read = fc_in.read(comma_buffer);
+            comma_buffer.flip();
+            Short delim = comma_buffer.getShort();
+            comma_buffer.flip();
+            if (display) {
+                System.out.format("delimiter: %04x\n", delim);
+            }
+            if (delim != 0x4c45) {
```

Comment:
Ideally this would be inherited from a common source but I think it is just a static in one of the files that reads logs...